### PR TITLE
Make Slack notifications filterable

### LINKED ISSUE #35: Avoid Slack notifications for imports without changes
When nothing has changes you get the following notification:

> Project Name: *0* new strings were added, *0* were fuzzied, and *0* were obsoleted. There were *0* errors

IMO we should suppress such messages in https://github.com/wearerequired/traduttore/blob/0e9e7f63d4a13a6e626be4f9ee2da525e8325a59/inc/Plugin.php#L112-L130.

### PR DESCRIPTION
**Pull request by swissspidy**
Fixes #35.

### CODE REVIEW

File: inc/Plugin.php [hunk 1]
```diff
@@ -96,13 +96,37 @@ public function register_hooks(): void {
...
+						/**
+						 * Filters whether a Slack notification for translation updates from GitHub should be sent.
+						 *
+						 * @since 2.0.3
+						 *
+						 * @param bool               $send_message    Whether to send a notification or not. Default true.
+						 * @param GP_Translation_Set $translation_set Translation set the ZIP is for.
+						 * @param GP_Project         $project         The GlotPress project that was updated.
+						 */
+						$send_message = apply_filters( 'traduttore_zip_generated_send_notification', true, $translation_set, $project );
```
Comment (ocean90) [inline inc/Plugin.php]: Don't we want to separate the "namespace" and the action with a dot? `traduttore.zip_generated_send_notification`

File: inc/Plugin.php [hunk 2]
```diff
@@ -96,13 +96,37 @@ public function register_hooks(): void {
...
 						);
+
+						/**
+						 * Filters the Slack notification message when a new translation ZIP file is built.
+						 *
+						 * @since 2.0.3
+						 *
+						 * @param string $message The notification message.
+						 */
+						return apply_filters( 'traduttore_zip_generated_notification_message', $message );
```
Comment (ocean90) [inline inc/Plugin.php]: Should probably pass the same arguments as for the default $message or ` $zip_path, $zip_url, $translation_set`.

File: inc/Plugin.php [hunk 3]
```diff
@@ -127,6 +169,15 @@ public function register_hooks(): void {
...
 						);
+
+						/**
+						 * Filters the Slack notification message when new translations are updated from GitHub.
+						 *
+						 * @since 2.0.3
+						 *
+						 * @param string $message The notification message.
+						 */
+						return apply_filters( 'traduttore_updated_from_github_notification_message', $message );
```
Comment (ocean90) [inline inc/Plugin.php]: Same as for the other message, pass additional arguments to the filter.

File: inc/Plugin.php (hunk 1 above)
Comment (swissspidy) [reply inc/Plugin.php]: I hesitated to do so because the two existing actions we have aren't using a dot.

Comment (swissspidy) [reply inc/Plugin.php]: We could use the dot in the existing actions and use `do_action_deprecated` for the non-dot version. Then, we could use the dot for the newer ones, too.

Easier to do so now than later :-)

**Pull request merged**